toolbox/db/db_get.m

```diff
@@ -218,32 +218,58 @@
         else
             varargout{1} = sAnatFiles;
         end
-
-    % Usage: sFiles = db_get('FunctionalFile', FileType, FileIDs)
-    % Usage: sFiles = db_get('FunctionalFile', FileType, FileNames)
+        
+%% ==== FUNCTIONAL FILE ====
+    % Usage: [sFiles, sItems] = db_get('FunctionalFile', FileIDs,   Fields)
+    % Usage: [sFiles, sItems] = db_get('FunctionalFile', FileNames, Fields)
+    % Usage: [sFiles, sItems] = db_get('FunctionalFile', CondQuery, Fields)
     case 'FunctionalFile'
-        type = args{1};
-        iFiles = args{2};
+        % Parse inputs
+        iFiles = args{1};
+        fields = '*';                              
+        templateStruct = db_template('FunctionalFile');
+
         if ischar(iFiles)
             iFiles = {iFiles};
+        elseif isstruct(iFiles)
+            condQuery = CondQuery;           
         end
-        nFiles = length(iFiles);
-        sFiles = repmat(db_template('FunctionalFile'), 1, nFiles);
-        sItems = repmat(db_template(type), 1, nFiles);
 
+        if length(args) > 1
+            fields = args{2};
+            if ischar(fields)
+                fields = {fields};
+            end
+            for i = 1 : length(fields)
+                resultStruct.(fields{i}) = templateStruct.(fields{i});
+            end
+        else
+            resultStruct = templateStruct;
+        end
+              
+        nFiles = length(iFiles);
+        sFiles = repmat(resultStruct, 1, nFiles);
+        sItems= [];
+        
         for i = 1:nFiles
             if iscell(iFiles)
-                condQuery = struct('Type', type, 'FileName', iFiles{i});
+                condQuery.FileName = iFiles{i};
             else
-                condQuery = struct('Id', iFiles(i));
+                condQuery.Id = iFiles(i);
             end
-            sFiles(i) = sql_query(sqlConn, 'select', 'functionalfile', '*', condQuery);
-            sItems(i) = getFuncFileStruct(type, sFiles(i));
+            sFiles(i) = sql_query(sqlConn, 'select', 'functionalfile', fields, condQuery);
+            if strcmp(fields, '*')
+                if i == 1
+                    sItems = repmat(db_template(sFiles(1).Type), 1, nFiles);
+                end
+                sItems(i) = getFuncFileStruct(sFiles(i).Type, sFiles(i));
```

Comment:
Now that this function will be called often, it might be worth adding a condition on nargout > 1 to avoid creating sItems when only 1 argument is requested.

Reply:
Also, it's not guaranteed that the sql_query() will return the Type field. Make sure to add it to the list of fields when a second output is requested.

Reply:
Updated in daed99c
Returns the parsed FunctionalFiles when they are: expected, all are the same type and all fields were requested